acs-integration-tests/src/test/java/com/ge/predix/cloud/foundry/application/ApplicationHelper.java

```
@@ -0,0 +1,353 @@
+                        .host(applicationName)
+                        .domain(System.getProperty("CF_BASE_DOMAIN"))
+                        .buildpack("java-buildpack")
+                        .diskQuota(2048)
+                        .healthCheckType(ApplicationHealthCheck.PORT)
+                        .memory(2048)
+                        .name(applicationName)
+                        .noStart(true)
+                        .build())
+                .doOnError(onError -> LOGGER.error("Unable to push application '{}'", applicationName));
+    }
+
+    void setEnvironmentVariables(final String applicationName, final Map<String, String> environmentVariables)
+            throws InterruptedException {
+
+        setEnvironmentVariables(this.cloudFoundryOperations, applicationName, environmentVariables)
+                .blockLast();
+    }
+
+    private static Flux<Void> setEnvironmentVariables(final CloudFoundryOperations cloudFoundryOperations,
+            final String applicationName, final Map<String, String> environmentVariables) {
+
+        return Flux.fromIterable(environmentVariables.entrySet())
+                .flatMap(environmentVariable -> setEnvironmentVariable(cloudFoundryOperations, applicationName,
+                        environmentVariable.getKey(), environmentVariable.getValue()), 1);
+    }
+
+    private static Mono<Void> setEnvironmentVariable(final CloudFoundryOperations cloudFoundryOperations,
+            final String applicationName, final String variableName, final String variableValue) {
+
+        return cloudFoundryOperations.applications()
+                .setEnvironmentVariable(SetEnvironmentVariableApplicationRequest.builder()
+                        .name(applicationName)
+                        .variableName(variableName)
+                        .variableValue(variableValue)
+                        .build())
+                .doOnError(onError -> LOGGER.error("Unable to set environment variable '{}': '{}' on '{}'",
```

Should fail maven verify when this fails instead of continuing on to other tests

---

In theory this should fail the suite but maybe the `Throwable` is getting lost in translation. Will have to write a meta-test to actually force an in-flux exception and verify.